### PR TITLE
missing header include in SortImpl.cu

### PR DESCRIPTION
**Pull request by jeffdaily**
SortImpl.cu needs to include <thrust/execution_policy.h> for
thrust::host.  Depending on the nvidia/thrust or rocThrust version,
transitive inclusion of this header is not guaranteed.

### CODE REVIEW

**Review by peterbell10 — APPROVED** [review]

**Comment by jeffdaily** [comment]
I thought @peterbell10 had the magic authority to approve for the sake of pytorchbot.  :-)  I'll wait for @albanD's approval then.

I got this quote from the rocThrust team why this change will be needed in the upcoming ROCm 5.2 release.

> The transitive inclusion of thrust/execution_policy.h (in which thrust::host is defined) has changed in nvidia/thrust and rocThrust the following way. Y stands for included, whereas N stands for not included.

Header | nvidia/thrust 1.15 | rocThrust 1.15 | nvidia/thrust 1.16 | rocThrust 1.16
-- | -- | -- | -- | --
copy.h | Y | Y | N | N
sequence.h | Y | N | N | N
sort.h | Y | Y | Y | N
device_ptr.h | N | N | N | N
scan.h | Y | Y | N | N

> As can be seen, neither was the inclusion consistent between rocThrust and nvidia/thrust before the update, nor nvidia/thrust keeps transitive include hierarchies intact in their minor version updates. In fact, this set of include files would still include thrust/execution_policy.h using nvidia/thrust after the update, but it is more of a happy accident than anything deliberate.

> Because of this, I would suggest to any consumer of these libraries not to depend on transitive includes, but include the used headers directly in their source files.

**Review by albanD — APPROVED** [review]
Thanks for the details.

**Pull request closed without merging**
